tests/runtime/test_flow_run.py

```diff
@@ -28,6 +29,15 @@ async def test_attribute_override_via_env_var(self, monkeypatch):
         monkeypatch.setenv(name="PREFECT__RUNTIME__FLOW_RUN__NEW_KEY", value="foobar")
         assert flow_run.new_key == "foobar"
 
+    async def test_scheduled_start_time_override_via_env_var(self, monkeypatch):
+        # this env var must be converted to pendulum.DateTime
+        dt = "2023-05-13 20:00:00"
+        monkeypatch.setenv(
+            name="PREFECT__RUNTIME__FLOW_RUN__SCHEDULED_START_TIME", value=dt
+        )
+        assert isinstance(flow_run.scheduled_start_time, pendulum.DateTime)
+        assert flow_run.scheduled_start_time == pendulum.instance(dateparser.parse(dt))
+
```

Comment:
It would be ideal to have tests in:
`test_task_run.py` and `test_deployment.py` as well!



It makes sense to test all different types, perhaps using pytest's parametrize like so:

https://github.com/PrefectHQ/prefect/blob/d709b49c587053423eb61d05ad0c261d0adbadc7/benches/bench_flows.py#L30-L33

Reply:
That's also a good idea! I implemented the tests for all the known attributes, including those whose cannot be mocked (and checking that it raises a ValueError).

I left two attributes from be tested in `test_deployment.py`, because they are Optional[Dict] types, and by default they are None during the test executions.